### PR TITLE
Feature/abandoned cart tracking

### PR DESCRIPTION
**Pull request by mrogers-5s**

### CODE REVIEW

File: lib/spree/chimpy/interface/order_upserter.rb [hunk 1]
```diff
@@ -40,53 +33,29 @@ def find_and_update_order(data)
...
-          tax_total:         @order.try(:included_tax_total).to_f + @order.try(:additional_tax_total).to_f,
-          customer: {
-            id: customer_id
-          }
-        }
-
-        if source
-          data[:campaign_id] = source.campaign_id
+        begin
+          store_api_call.carts(@order.number).delete
```
Comment (Ken-Michalak) [inline lib/spree/chimpy/interface/order_upserter.rb]: I would expect this to go along with the `send_..._to_mail_chimp` that's currently in the order_decorator. So like, when the order is completed, delete the cart, and create the order.

Comment (mrogers-5s) [reply lib/spree/chimpy/interface/order_upserter.rb]: Wouldn't we want to keep direct API calls out of the model classes, though?. The upserter (in the base repo, anyway) seems to be acting like a middle man between the model object and the API

`Spree::Chimpy.enqueue()` effectively sets up a delayed call to the upserter classes.

Would it be better if we used something like `enqueue_upsert()` and `enqueue_delete()`, or am I overthinking it?

File: app/models/spree/order_decorator.rb [hunk 2]
```diff
@@ -22,6 +17,6 @@ def send_cart_to_mail_chimp
   def handle_cancelation
     canceled = state_changed? && canceled?
     yield
-    send_order_to_mail_chimp if canceled
+    snotify_mail_chimp if canceled
```
Comment (Ken-Michalak) [inline app/models/spree/order_decorator.rb]: typo

File: app/models/spree/order_decorator.rb [hunk 3]
```diff
@@ -1,19 +1,14 @@
...
-  def send_cart_to_mail_chimp
-    unless (self.completed?)
-      # If this Order is completed, do not add a Cart to MailChimp
+  def notify_mail_chimp
+    if (self.completed?)
+      # If this Order is complete, send an Order to MailChimp & Remove any existing Carts
+      Spree::Chimpy.enqueue(:order, self) if Spree::Chimpy.configured?
+    else
+      # If this Order is NOT completed, add/update the Cart entry in MailChimp
       Spree::Chimpy.enqueue(:cart, self) if Spree::Chimpy.configured?
```
Comment (Ken-Michalak) [inline app/models/spree/order_decorator.rb]: With the delete cart moved, we want to sync the cart even when the order is completed.

**Comment by jspizziri** [comment]
@mrogers-5s please squash these commits. We only want this feature to show up as a single commit so that we eliminate commit noise.

**Comment by jspizziri** [comment]
@mrogers-5s there are merge conflicts with master. Please fix and then I'll merge.

**Pull request merged**